src/main/java/com/ft/jerseyhttpwrapper/providers/HostAndPortIpResolver.java

```diff
@@ -34,12 +33,13 @@ public HostAndPortIpResolver(HostToIpMapper hostMapper) {
 	}
 
 	public List<HostAndPort> resolve(HostAndPort hostAndPort) {
+		final Operation resultOperation = operation("resolve").with("argument", hostAndPort.getHost()).initiate(this);
 		try {
-			final InetAddress[] inetAddresses = hostMapper.mapToIps(hostAndPort.getHostText());
+			final InetAddress[] inetAddresses = hostMapper.mapToIps(hostAndPort.getHost());
 			return mapAddressesAcrossPorts(hostAndPort, inetAddresses);
 
 		} catch(UnknownHostException e) {
-			logger.warn("Unable to resolve host {}", hostAndPort.getHostText());
+			resultOperation.wasFailure().logWarn("Unable to resolve host"  + hostAndPort.getHost());
```

Comment:
Need to test if this will output JSON without specifically making the text in JSON format, for example: `operation.wasSuccessful().yielding("result","{\"text\": \"hello world\"}").log();`